src/Catalog/Registration/RegistrationCollector.cs

```diff
@@ -57,7 +57,7 @@ protected override Task<IEnumerable<CatalogItemBatch>> CreateBatches(IEnumerable
                 .GroupBy(item => GetKey(item.Value))
                 .Select(group => new CatalogItemBatch(
                     group.Min(item => item.CommitTimeStamp),
-                    group));
```

Comment:
Add the reason why we are ordering the batch in the gigantic comment above.

E.g.
```
// Order the batch so that catalog items are processed in the order that they occurred.
// Otherwise, several simultaneous changes can result in them being processed in the wrong order.
```